lib/synapse/service_watcher/base.rb

```diff
@@ -117,7 +117,7 @@ def set_backends(new_backends)
       # Aggregate and deduplicate all potential backend service instances.
       new_backends = (new_backends + @default_servers) if @keep_default_servers
       new_backends = new_backends.uniq {|b|
-        [b['host'], b['port'], b.fetch('name', '')]
+        [b['host'], b['port'], b.fetch('name', ''), b.fetch('weight', 1)]
```

Comment:
we should probably either unique on a strictly limited set of properties (just host, port, possibly name) or on all of the possible properties.